File: jollyJumpers/jollyJumpers.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    long n;
    while (cin >> n) {
        vector<long> v(n);
        bool jolly = true;
        for (int i = 0; i < v.size(); i++) {
            cin >> v[i];
        }

        vector<bool> seen(n, false);

        for(int i = 1; i < n; i++) {
            long diff = abs(v[i] - v[i-1]);
            if(diff >= 0 && diff < n) {
                seen[diff] = true;
            }
        }

        for(int i = 1; i < n; i++) {
            jolly &= seen[i];
        }

        if (jolly) {
            cout << "Jolly" << endl;
        } else {
            cout << "Not jolly" << endl;
        }
    }
}

File: jollyJumpers/jollyJumpers.py
a = list(map(int,input().split()))
val = a[0] - 1


for i in range(1, len(a) - 1):
    if (abs(a[i] - a[i+1]) != val):
        print("Not jolly")
        exit()
    val -= 1
print("Jolly")